Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,317 @@
+            }
+            Send();
+        }
+
+        /// <summary>
+        /// Force BacktraceSession to Send data to Backtrace
+        /// </summary>
+        public void Send()
+        {
+            Send(numberOfRetries: 3);
+        }
+
+        /// <summary>
+        /// Force BacktraceSession to Send data to Backtrace with retry settings
+        /// </summary>
+        public void Send(int numberOfRetries = DefaultNumberOfRetries)
+        {
+            if (numberOfRetries == 0)
+            {
+                Debug.LogWarning("Backtrace Session: Cannot send session data to Backtrace due to submission issue.");
+                CleanEvents();
+                return;
+            }
+            if (Count() == 0)
+            {
+                return;
+            }
+            // create a copy of events 
+            var uniqueEvents = UniqueEvents.ToArray();
+            var sessionEvents = SessionEvents.ToArray();
+            var payload = CreateJsonPayload(uniqueEvents, sessionEvents);
+
+            // cleanup existing copy of events
+            CleanEvents();
+
+            // submit data to Backtrace
+            RequestHandler.Post(SubmissionUrl, payload, (long statusCode, bool httpError, string response) =>
+            {
+                if (statusCode == 200)
+                {
+                    OnRequestCompleted();
+                }
+                else
+                {
+                    OnRequestFailure(uniqueEvents, sessionEvents);
+                    // assume that we should try to retry request on 503
+                    Send(numberOfRetries - 1);
+                }
+            });
+        }
+
+        /// <summary>
+        /// Add unique event to next Backtrace session request
+        /// </summary>
+        /// <param name="attributeName">attribute name</param>
+        public bool AddUniqueEventAttribute(string attributeName, IDictionary<string, string> attributes = null)
+        {
+            if (!ShouldProcessEvent(attributeName))
+            {
+                Debug.LogWarning("Skipping report: Reached store limit or event has empty name.");
+                return false;
+            }
+            // add to event attributes, attribute provider attributes
+            if (attributes == null)
+            {
+                attributes = new Dictionary<string, string>();
+            }
+            _attributeProvider.AddAttributes(attributes);
+
+            // validate if unique event attribute is available and
+            // prevent undefined attributes
+            if (attributes.TryGetValue(attributeName, out string attributeValue) == false || string.IsNullOrEmpty(attributeValue))
+            {
+                Debug.LogWarning("Attribute name is not available in attribute scope. Please define attribute to set unique event.");
+                return false;
+            }
+            var existingEvent = UniqueEvents.FirstOrDefault(m => m.Name == attributeName);

```

Comment:
The actual logic for this needs to check that all attributes are equal for both events, not just that names match.  For unique, sending two copy of the exactly duplicate event doesn't make sense, but all other cases should likely be allowed (e.g. what if we're using a linked attribute level, and it's changing all the time).

I want to suggest not trying to guard against this at all is the right choice both for unique and for session.

Reply:
Removed unique check - somehow I though we should send only one unique/session event with specific name. Thanks for pointing this out!